File: engine/model/src/main/scala/tools/packet/fredrick/GetFredrickInfo.scala
package tools.packet.fredrick

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GetFredrickInfo(private var _characterId: Int, private var _merchantNetMeso: Int) extends PacketInput {
  def characterId: Int = _characterId

  def merchantNetMeso: Int = _merchantNetMeso

  override def opcode(): SendOpcode = SendOpcode.FREDRICK
}

File: engine/engine-model/src/main/java/tools/packet/fredrick/GetFredrickInfo.java
package tools.packet.fredrick;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GetFredrickInfo(Integer characterId, Integer merchantNetMeso) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.FREDRICK;
   }
}